tests/unit_tests/commands/report/base_test.py

```diff
@@ -0,0 +1,191 @@
+    alert_minimum_interval: int | None = None,
+    report_minimum_interval: int | None = None,
+) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
+    """
+    Decorator to mock the current_app.config values dynamically for each test.
+
+    :param alert_minimum_interval: Minimum interval in minutes for alerts. Defaults to None.
+    :param report_minimum_interval: Minimum interval in minutes for reports. Defaults to None.
+
+    :returns: A decorator that wraps a function.
+    """
+
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
+        @wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            with patch(
+                "superset.commands.report.base.current_app.config"
+            ) as mock_config:
+                mock_config.get.side_effect = lambda key: {
+                    "ALERT_MINIMUM_INTERVAL_MINUTES": alert_minimum_interval,
+                    "REPORT_MINIMUM_INTERVAL_MINUTES": report_minimum_interval,
+                }.get(key)
+                return func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+@app_custom_config()
+def test_validate_report_frequency() -> None:
+    """
+    Test the ``validate_report_frequency`` method when there's
+    no minimum frequency configured.
+    """
+    for report_type in REPORT_TYPES:
+        for schedule in TEST_SCHEDULES:
+            BaseReportScheduleCommand().validate_report_frequency(
+                schedule,
+                report_type,
+            )
+
+
+@app_custom_config(alert_minimum_interval=4, report_minimum_interval=5)
+def test_validate_report_frequency_minimum_set() -> None:
+    """
+    Test the ``validate_report_frequency`` method when there's
+    minimum frequencies configured.
+    """
+
+    BaseReportScheduleCommand().validate_report_frequency(
+        "1,5 * * * *",
+        ReportScheduleType.ALERT,
+    )
+    BaseReportScheduleCommand().validate_report_frequency(
+        "6,11 * * * *",
+        ReportScheduleType.REPORT,
+    )
+
+
+@app_custom_config(alert_minimum_interval=2, report_minimum_interval=5)
+def test_validate_report_frequency_invalid_schedule() -> None:
+    """
+    Test the ``validate_report_frequency`` method when the configured
+    schedule exceeds the limit.
+    """
+    with pytest.raises(ReportScheduleFrequencyNotAllowed):
+        BaseReportScheduleCommand().validate_report_frequency(
+            "1,2 * * * *",
+            ReportScheduleType.ALERT,
+        )
+
+    with pytest.raises(ReportScheduleFrequencyNotAllowed):
+        BaseReportScheduleCommand().validate_report_frequency(
+            "1,5 * * * *",
+            ReportScheduleType.REPORT,
+        )
+
+
+@app_custom_config(alert_minimum_interval=10)
+def test_validate_report_frequency_alert_only() -> None:
+    """
+    Test the ``validate_report_frequency`` method when there's
+    only a configuration for alerts and user is creating report.
+    """
+    for schedule in TEST_SCHEDULES:
```

Comment:
I think @pytest.mark.parametrize would work well for you on these for loop tests.

Reply:
ohh, wasn't familiar with it! just added it